test/test_python_type_stubs.py

```diff
@@ -0,0 +1,332 @@
+    Nullable,
+    UInt64,
+    Union,
+    UnionField,
+    Void,
+    Timestamp, Alias)
+from stone.target.python_type_stubs import PythonTypeStubsGenerator
+
+ISO_8601_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+
+def _make_generator():
+    # type: () -> PythonTypeStubsGenerator
+    generator = PythonTypeStubsGenerator(
+        target_folder_path=Mock(),
+        args=Mock()
+    )
+    return generator
+
+def _mock_emit(generator):
+    # type: (PythonTypeStubsGenerator) -> List[str]
+    """
+    Mock out PythonTypeStubsGenerator's .emit function, and return a list containing all params
+    emit was called with.
+    """
+    recorded_emits = []  # type: List[str]
+
+    def record_emit(s):
+        recorded_emits.append(s)
+
+    orig_append = generator._append_output
+    generator._append_output = Mock(wraps=orig_append, side_effect=record_emit)  # type: ignore
+
+    return recorded_emits
+
+def _make_namespace_with_alias():
+    # type: (...) -> ApiNamespace
+    ns = ApiNamespace('ns_with_alias')
+
+    struct1 = Struct(name='Struct1', namespace=ns, token=None)
+    struct1.set_attributes(None, [StructField('f1', Boolean(), None, None)])
+    ns.add_data_type(struct1)
+
+    alias = Alias(name='AliasToStruct1', namespace=ns, token=None)
+    alias.set_attributes(doc=None, data_type=struct1)
+    ns.add_alias(alias)
+
+    return ns
+
+def _make_namespace_with_many_structs():
+    # type: (...) -> ApiNamespace
+    ns = ApiNamespace('ns_with_many_structs')
+
+    struct1 = Struct(name='Struct1', namespace=ns, token=None)
+    struct1.set_attributes(None, [StructField('f1', Boolean(), None, None)])
+    ns.add_data_type(struct1)
+
+    struct2 = Struct(name='Struct2', namespace=ns, token=None)
+    struct2.set_attributes(
+        doc=None,
+        fields=[
+            StructField('f2', StoneList(UInt64()), None, None),
+            StructField('f3', Timestamp(ISO_8601_FORMAT), None, None)
+        ]
+    )
+    ns.add_data_type(struct2)
+
+    return ns
+
+def _make_namespace_with_nested_types():
+    # type: (...) -> ApiNamespace
+    ns = ApiNamespace('ns_w_nested_types')
+
+    struct = Struct(name='NestedTypes', namespace=ns, token=None)
+    struct.set_attributes(
+        doc=None,
+        fields=[
+            StructField(
+                name='NullableList',
+                data_type=Nullable(
+                    StoneList(UInt64())
+                ),
+                doc=None,
+                token=None,
+            ),
+            StructField(
+                name='ListOfNullables',
+                data_type=StoneList(
+                    Nullable(UInt64())
+                ),
+                doc=None,
+                token=None,
+            )
+        ]
+    )
+    ns.add_data_type(struct)
+
+    return ns
+
+def _make_namespace_with_a_union():
+    # type: (...) -> ApiNamespace
+    ns = ApiNamespace('ns_with_a_union')
+
+    u1 = Union(name='U1', namespace=ns, token=None, closed=True)
+    u1.set_attributes(
```

Comment:
We need a union that has a variant with type other than void. Non-void variants will generate an additional `get_variant()` function along with the `is_variant()` that you already type.